### PR TITLE
New readiness checklist

### PR DESCRIPTION
**Pull request by FabrizioMoggio**
#### What type of PR is this?
updating the readiness checklist according to the new format define in Release Management (https://github.com/camaraproject/ReleaseManagement/blob/25400e8f484dbb6e53a7dcc21d3a709b6cc2b686/documentation/API-Readiness-Checklist.md)

Add one of the following kinds:
* documentation

#### What this PR does / why we need it:
To be coherent with the readiness checklist

#### Which issue(s) this PR fixes:

https://github.com/camaraproject/CallForwardingSignal/issues/34

### CODE REVIEW

**Comment by FabrizioMoggio** [comment]
> Let's wait we're aligned on #42 and this one will be good :)

you are totally right :-)

**Review by bigludo7 — DISMISSED** [review]
We will have to provide links for some of the assets but I guess we can do it later.

**Comment by FabrizioMoggio** [comment]
> We will have to provide links for some of the assets but I guess we can do it later.

once I have produced the release tag we are limited in making change. let me add the links

**Comment by rartych** [comment]
> We will have to provide links for some of the assets but I guess we can do it later.

We can use relative links, that should work with release tags.
https://github.blog/2013-01-31-relative-links-in-markup-files/

**Comment by FabrizioMoggio** [comment]
> > We will have to provide links for some of the assets but I guess we can do it later.
> 
> We can use relative links, that should work with release tags. https://github.blog/2013-01-31-relative-links-in-markup-files/

I was thinkig of using the same approach as QoD: https://github.com/camaraproject/QualityOnDemand/blob/main/documentation/API_documentation/quality-on-demand-API-Readiness-Checklist.md

not links but a reference to the path

File: documentation/API_documentation/CFS-API-Readiness-Checklist.md [hunk 1]
```diff
@@ -1,13 +1,20 @@
...
-|  2 |API Implementation                |   N       | --                     |
-|  3 |API Documentation                 |   Y       | Contributed                |
-|  4 |User Stories                      |   Y       | Approved               |
-|  5 |API test cases                    |   Y       | Pending                |
-|  6 |Validated by atleast 2 operators  |   Y       | Pending                | 
-|  7 |Security review                   |   Y       | Pending                |
+
+# API Readiness Checklist
+
+Checklist for CFS API wip
```
Comment (hdamker) [inline documentation/API_documentation/CFS-API-Readiness-Checklist.md:4]:
```suggestion
Checklist for API call-forwarding-signal v0.2.0-rc.1 in r1.1

```

File: documentation/API_documentation/CFS-API-Readiness-Checklist.md [hunk 2]
```diff
@@ -1,20 +1,21 @@
...
 |  3 | Guidelines from ICM applied                  |   O   |         M         |    M    |    M   |   Y  |      |
-|  4 | API versioning convention applied            |   M   |         M         |    M    |    M   |   Y  |      |
+|  4 | API versioning convention applied            |   M   |         M         |    M    |    M   |   Y  | v0.2.0-rc.1 |
 |  5 | API documentation                            |   M   |         M         |    M    |    M   |   Y  | inline in YAML |
-|  6 | User stories                                 |   O   |         O         |    O    |    M   |   Y  |      |
-|  7 | Basic API test cases & documentation         |   O   |         M         |    M    |    M   |   N  | /documentation/API_documentation/CFS_UserStory_UseCase-1.md |
+|  6 | User stories                                 |   O   |         O         |    O    |    M   |   Y  | /documentation/API_documentation/CFS_UserStory_UseCase-1.md |
+|  7 | Basic API test cases & documentation         |   O   |         M         |    M    |    M   |   N  | code/Test_definitions/call-forwarding-signal-every-forwarding.feature, code/Test_definitions/call-forwarding-signal-unconditional.feature |
```
Comment (hdamker) [inline documentation/API_documentation/CFS-API-Readiness-Checklist.md:15]:
```suggestion
|  7 | Basic API test cases & documentation         |   O   |         M         |    M    |    M   |   N  | /code/Test_definitions/call-forwarding-signal-every-forwarding.feature, /code/Test_definitions/call-forwarding-signal-unconditional.feature |
```

Also these links are relative to the repository root (like the others).

BTW: https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax#relative-links

**Review by bigludo7 — APPROVED** [review]
LGTM

**Pull request merged**